app/react/translations/edit/editors/BooleanEditor.js

```diff
@@ -1,64 +1,81 @@
 /* @flow */
-import React, { Component } from 'react';
+import Radium from 'radium';
+import React, { PureComponent } from 'react';
+
+import Checkbox from '../../../components/Checkbox.react';
+import EditorSave from './EditorSave';
+import EditorWrapper from './EditorWrapper';
+import UnsavedLabel from './UnsavedLabel';
 
 import type { FieldInfo } from '../index';
 
 type PropTypes = {
+  focused: boolean,
+  fieldInfo: FieldInfo,
+  handleBlur: Function,
+  handleFocus: Function,
   onChange: Function,
   onSubmit: Function,
-  value: string,
+  registerTabPress: Function,
   saved: boolean,
-  fieldInfo: FieldInfo
+  tabPressed: ?boolean,
+  value: string,
 };
 
-export default class BooleanEditor extends Component {
-  toggleRadio = () => {
-    const { onChange, value, fieldInfo } = this.props;
-    const newValue = value === 'true' ? 'false' : 'true';
+@Radium
+export default class BooleanEditor extends PureComponent {
+  onCheckboxChange = (newValue) => {
+    const { onChange, fieldInfo } = this.props;
     onChange(newValue, fieldInfo);
   }
 
   props: PropTypes
 
+  handleBlur = () => {
+    if (this.props.tabPressed) {
+      this.handleSubmit();
```

Comment:
Propagate blur event to handleSubmit and remove check for event presence in handleSubmit method.